lib/active_encode/engine_adapters/ffmpeg_adapter.rb

```diff
@@ -206,7 +212,11 @@ def ffmpeg_command(input_url, id, opts)
           file_name = "outputs/#{sanitized_filename}-#{output[:label]}.#{output[:extension]}"
           " #{output[:ffmpeg_opt]} #{working_path(file_name, id)}"
         end.join(" ")
-        "#{FFMPEG_PATH} -y -loglevel error -progress #{working_path('progress', id)} -i \"#{input_url}\" #{output_opt}"
+        header_opt = opts[:headers]&.map do |k, v|
+          "-headers '#{k}: #{v}'"
+        end&.join(" ")
```

Comment:
This might be a little cleaner if you force it into an array to start with:
```suggestion
        header_opt = Array(opts[:headers]).map do |k, v|
          "-headers '#{k}: #{v}'"
        end.join(" ")
```

Reply:
Cool trick